NumberTheory/FastPower.cpp

```diff
@@ -0,0 +1,47 @@
+/*
+<Fast Power>
+--------------
+Fast Power is an optimized algorithm to compute exponentiation in a short time.
+
```

Comment:
It would be helpful for others if you could also mention in 1 or 2 lines more what the algorithm does.